src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/CSharpCodeParser.cs

```diff
@@ -2596,10 +2630,11 @@ private IReadOnlyList<SyntaxToken> SkipToNextImportantToken(in SyntaxListBuilder
             }
             else
             {
-                return whitespace;
+                return;
             }
+
+            whitespace.Clear();
```

Comment:
Should `whitespace.Clear()` be called also outside the loop?

Reply:
There isn't a need, since this line is guaranteed to be hit before the loop exits.

Reply:
If `EndOfFile` is `true` at the start of this method, the loop isn't even entered. That cannot happen?

Reply:
Ah, you're right. Thank you! I wasn't looking at it correctly. Will update!

Reply:
Finally getting to this feedback. :smile:

Taking a closer look, it isn't necessary to call `Clear()` here. However, the `PooledArrayBuilder<SyntaxToken>` is expected to be empty so that `Clear()` would be a no-op if the loop isn't entered. To make this clearer, I'll add asserts throughout the PR to verify that the builders that are passed in are empty when they're expected to be filled.